azurerm/helpers/azure/app_service.go

```diff
@@ -212,6 +411,346 @@ func SchemaAppServiceSiteConfig() *schema.Schema {
 	}
 }
 
+func ExpandAppServiceAuthSettings(input interface{}) web.SiteAuthSettingsProperties {
+	settings := input.([]interface{})
```

Comment:
can we make this cast prior to passing into this method?

Reply:
I've checked how we call other expand functions and we don't seem to explicitly cast before calling them. Would you still like me to do that for this case?

Another example where we perform the cast inside the expand function is `ExpandAppServiceSiteConfig`

Reply:
that's (either passing in `d` and then expanding internally, or passing in an element and then converting it internally) an older pattern in the provider - for newer resources we're switching to doing:

```
fooRaw := d.Get("foo").([]interface{})
foo := expandFoo(fooRaw)

...

func expandFoo(input []interface{}) *foo.Bar {
  return &foo.Bar{}
}
```

Reply:
@tombuildsstuff Ahh okay, thank you for clarifying. I'll fix it ASAP! 👍